src/actions/addCartList.ts

```diff
@@ -0,0 +1,62 @@
+"use server";
+
+import { revalidatePath } from "next/cache";
+import { cookies } from "next/headers";
+
+import { createServerActionClient } from "@supabase/auth-helpers-nextjs";
+
+import { prisma } from "../lib/prisma";
+import { Database } from "../types/SupabaseTypes";
+
+const createCartListItem = async (cartListId: number, ingredientIds: number[]) => {
+  await prisma.cartListItem.createMany({
+    data: ingredientIds.map((ingredientId) => ({
+      cartListId,
+      ingredientId,
+    })),
+  });
+};
+
+export const addCartList = async (recipeId: string, ingredientIds: number[]) => {
+  const supabaseServerClient = createServerActionClient<Database>({ cookies });
+
+  const {
+    data: { session },
+  } = await supabaseServerClient.auth.getSession();
+
+  if (!session) {
+    throw new Error("認証に失敗しました");
+  }
+
+  const cartList = await prisma.cartList.findMany({
+    include: {
+      CartListItem: true,
+    },
+    orderBy: {
+      displayOrder: "desc",
+    },
+  });
+
+  const foundCartList = cartList.find((c) => c.recipeId === recipeId);
```

Comment:
これだと一度cartListのテーブル全取得して、そこからレシピIDに一致するcartListを取得しているので不必要なデータ取得が含まれていると思います！

なので`findFirst`を使用して不必要なクエリ結果の探索を避けたいと思った次第ですー！

```suggestion
  const existingCartList = await prisma.cartList.findFirst({
    where: {
      userId: session.user.id,
      recipeId: recipeId,
    },
    include: {
      CartListItem: true,
    },
  });

```

Reply:
メモ:display orderの要件実現できるか確認

Reply:
レシピを1件検索、display_orderの大きいやつだけを取ってくる。